### DISCUSSION TITLE
GBAHD DIY troubleshoot

### DISCUSSION THREAD

**Discussion by RubyOtaku** [darthcloud/BlueRetro#925] [General]
### BlueRetro firmware version

v1.9.2_hw2

### BlueRetro firmware specification

HW2

### BlueRetro firmware variant

System specific

### BlueRetro hardware type

Internal install DIY

### Manufacturer

GBAHD ESP32

### System used

GBAHD

### Bluetooth controller brand & name

8BitDo SF30 Pro & Sony Dualsense

### What is problem? (only list ONE problem per report)

Bluetooth controllers not connecting.

### What did you expect to happen?

Bluetooth controllers to connect automatically once I started syncing them. Used X Input on SF30 Pro & pairing mode on Dualsense. I still have the L & R buttons on the GBA motherboard connected as I'm installing it back into a shell so I can use it for motion capable games, L & R buttons working as they should so I imagine the connections to the FPGA should be fine. But I can't test any Bluetooth controllers. As mentioned on Twitter I omitted step 6 as I wanted to power this via the USB C port, Power switch set to 5V & DIP Switch 5th bit into slave mode. Before testing with Putty I disconnected BT+ & BT- on the GBA so the ESP32 isn't supplying power to it.

**Comment by darthcloud** [comment]
Moved to discussion since not a bug.

Your log is complete garbage.

Did you set the baud rate properly?

If so double if you flashed the firmware properly.

**Reply by RubyOtaku** [reply]
Apologies! I followed the instructions on the putty guide as they were shown.

**Comment by darthcloud** [comment]
Use HW1 firmware not 2

**Reply by RubyOtaku** [reply]
Thanks, do I just flash the firmware as I did before, but use HW1 instead of 2? Then it will be flashed correctly?

**Reply by RubyOtaku** [reply]
Flashed with HW1 and still having the same issue. Checking the log from Putty I'm getting the same mess of code on my end.

**Reply by darthcloud** [reply]
take screenshots of the esp programmer after programming done, and also take screenshots of putty settings and post here.

**Comment by RubyOtaku** [comment]
![Screenshot 2024-02-28 012754](https://github.com/darthcloud/BlueRetro/assets/154464411/bbfa4898-3c70-450c-9b2c-0adb0675382f)
![Screenshot 2024-02-28 012853](https://github.com/darthcloud/BlueRetro/assets/154464411/464d57ab-fa1c-43a0-876c-f2c0bd6ca7dc)
![Screenshot 2024-02-28 012911](https://github.com/darthcloud/BlueRetro/assets/154464411/3e2c4c16-931c-4391-b518-2a565d4306bf)
![Screenshot 2024-02-28 012932](https://github.com/darthcloud/BlueRetro/assets/154464411/378157d8-4ce7-4cf9-b4c0-b4df049bdf9a)

**Reply by darthcloud** [reply]
If you disconnect GBA from power source, does the esp32 log show up?

**Reply by RubyOtaku** [reply]
I've disconnected the wires from the ESP32 to GBA before testing with Putty as it said in the Putty guide "DO NOT CONNECT the 3.3v or 5V pin coming from game system" Not used Putty with the GBA receiving power from the ESP32 at any time.

**Comment by darthcloud** [comment]
i got no idea, maybe post picture of how things are wired to sea board.

**Comment by RubyOtaku** [comment]
![Screenshot 2024-02-28 021231](https://github.com/darthcloud/BlueRetro/assets/154464411/76b2fab6-3fe3-4684-b0e8-40c0aa5e4606)
For reference also, these exact steps were omitted in case there was any miscommunication from my end.

![20240228_021727](https://github.com/darthcloud/BlueRetro/assets/154464411/38fb68da-0f82-4e0d-83d0-81764b44b839)
Wire from TP4 came loose whilst getting a good angle on the board.

![20240226_230121](https://github.com/darthcloud/BlueRetro/assets/154464411/2a20a6a2-a0f4-4252-b03f-6fbc67f871be)
Apologies for the blurry photo, I took it as a reference point for myself whilst soldering. Red is for BT+ & White is for BT- on the GBA.

**Comment by RubyOtaku** [comment]
Anything standing out?

**Comment by RubyOtaku** [comment]
If there's any other troubleshooting I can do please let me know :) Really want to get this build finished for myself

**Reply by darthcloud** [reply]
I got no idea ¯\_(ツ)_/¯

You need to figure why BlueRetro FW is not booting on your sea board or why your see garbage on serial.

**Reply by RubyOtaku** [reply]
I removed all wiring in the board, cleaned up all the points and checked for bridges, then re did the wiring minus what's required for the GBA connection, re installed all firmware as per all the guide. Tested with Putty and still get the exact same result. I'm so lost on this.
